src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ArrayOfEmbeddedPointersNode.cs

```diff
@@ -102,5 +117,22 @@ public string MangledName
                 }
             }
         }
+        
+        private class EmbeddedPointerIndirectionWithSymbolAndOnMarkedCallbackNode : EmbeddedPointerIndirectionWithSymbolNode
```

Comment:
If you put the extra logic into `EmbeddedPointerIndirectionWithSymbolNode`, it probably won't hurt too much (making the delegate an optional parameter). Or even better, into the base class of it.

I don't expect that logic to get much more complicated over time to require the extra class. But I'll leave this up to you.